### PR TITLE
feat: [AXIMST-166] certificates screen with data

### PR DESCRIPTION
**Pull request by khudym**
YT: https://youtrack.raccoongang.com/issue/AXIMST-166

### CODE REVIEW

File: src/certificates/certificates-card/certificate-details/CertificateDetails.jsx [hunk 1]
```diff
@@ -19,7 +26,26 @@ const CertificateDetails = ({
...
+                iconAs={Icon}
+                variant="primary"
+                tooltipContent={<div>{intl.formatMessage(messages.editTooltip)}</div>}
+                alt={intl.formatMessage(messages.editTooltip)}
+                // onClick={handleEditAll} TODO https://youtrack.raccoongang.com/issue/AXIMST-178
+              />
+              <IconButtonWithTooltip
+                src={DeleteOutlineIcon}
+                iconAs={Icon}
+                variant="primary"
```
Comment (PKulkoRaccoonGang) [inline src/certificates/certificates-card/certificate-details/CertificateDetails.jsx:42]: [important]: The primary variant is the default for this component

File: src/certificates/certificates-card/certificate-details/CertificateDetails.jsx [hunk 2]
```diff
@@ -19,7 +26,26 @@ const CertificateDetails = ({
       <Stack className="justify-content-between" direction="horizontal">
         <h2 className="lead">{intl.formatMessage(messages.detailsSectionTitle)}</h2>
         <Stack direction="horizontal" gap="2">
-          {/* {mode === MODE_STATES.VIEW && ()} TODO https://youtrack.raccoongang.com/issue/AXIMST-166 */}
+          {mode === MODE_STATES.VIEW && (
+            <>
+              <IconButtonWithTooltip
+                src={EditOutlineIcon}
+                iconAs={Icon}
+                variant="primary"
```
Comment (PKulkoRaccoonGang) [inline src/certificates/certificates-card/certificate-details/CertificateDetails.jsx:34]: [important]: The primary variant is the default for this component

File: src/certificates/certificates-card/certificate-signatories/signatory/Signatory.jsx [hunk 3]
```diff
@@ -106,8 +126,16 @@ const Signatory = ({
...
-        null
-        // <Stack direction="horizontal" gap="2" className="stack-container" data-testid="signatory-view" /> TODO https://youtrack.raccoongang.com/issue/AXIMST-166
+        <Stack direction="horizontal" gap="2" className="stack-container" data-testid="signatory-view">
+          <Stack className="text-stack">
+            <p><b>{intl.formatMessage(messages.nameLabel)}</b> {name}</p>
+            <p><b>{intl.formatMessage(messages.titleLabel)}</b> {title}</p>
+            <p><b>{intl.formatMessage(messages.organizationLabel)}</b> {organization}</p>
+          </Stack>
+          <div className="image-container">
+            {signatureImagePath && <img src={`${getConfig().STUDIO_BASE_URL}${signatureImagePath}`} alt="signature" className="d-block w-100" />}
```
Comment (PKulkoRaccoonGang) [inline src/certificates/certificates-card/certificate-signatories/signatory/Signatory.jsx:136]: [nit]: Let's add a translation for the alt text and try to use Paragon component Image

File: src/certificates/certificates-card/certificate-signatories/signatory/Signatory.test.jsx [hunk 4]
```diff
@@ -40,6 +40,13 @@ describe('Signatory Component', () => {
     expect(getByPlaceholderText(messages.namePlaceholder.defaultMessage)).toBeInTheDocument();
   });
 
+  it('renders in VIEW mode', () => {
+    const { getByText, queryByText } = renderSignatory({ ...defaultProps, mode: MODE_STATES.VIEW });
+    expect(getByText('John Doe')).toBeInTheDocument();
```
Comment (PKulkoRaccoonGang) [inline src/certificates/certificates-card/certificate-signatories/signatory/Signatory.test.jsx:45]: ` expect(defaultProps.name).toBeInTheDocument();
`

File: src/certificates/certificates-card/certificate-signatories/signatory/Signatory.test.jsx [hunk 5]
```diff
@@ -40,6 +40,13 @@ describe('Signatory Component', () => {
...
+    expect(getByText('Director')).toBeInTheDocument();
```
Comment (PKulkoRaccoonGang) [inline src/certificates/certificates-card/certificate-signatories/signatory/Signatory.test.jsx:46]: `expect(defaultProps.title).toBeInTheDocument();`

**Review by PKulkoRaccoonGang — CHANGES_REQUESTED** [review]

File: src/certificates/data/thunks.js [hunk 6]
```diff
@@ -21,3 +32,31 @@ export function fetchCertificates(courseId) {
...
+
+export function createCourseCertificate(courseId, certificates) {
+  return async (dispatch) => {
+    dispatch(updateSavingStatus({ status: RequestStatus.PENDING }));
+    dispatch(showProcessingNotification(NOTIFICATION_MESSAGES.saving));
+
+    try {
+      const certificatesValues = await createCertificate(courseId, certificates);
+      dispatch(createCertificateSuccess(certificatesValues));
+      dispatch(hideProcessingNotification());
```
Comment (monteri) [inline src/certificates/data/thunks.js:44]: move to `finally`

File: src/certificates/data/thunks.js [hunk 7]
```diff
@@ -21,3 +32,31 @@ export function fetchCertificates(courseId) {
...
+      let errorData;
+      try {
+        const { customAttributes: { httpErrorResponseData } } = error;
+        errorData = JSON.parse(httpErrorResponseData);
+      } catch (err) {
+        errorData = {};
+      }
+
+      dispatch(getDataSendErrors(errorData));
```
Comment (monteri) [inline src/certificates/data/thunks.js:57]: Check if this is required

File: src/certificates/hooks/useCertificates.jsx [hunk 8]
```diff
@@ -1,29 +1,49 @@
...
+
+  useEffect(() => {
+    if (!hasCertificateModes) {
+      dispatch(setMode(MODE_STATES.NO_MODES));
+    } else if (hasCertificateModes && !certificates.length) {
+      dispatch(setMode(MODE_STATES.NO_CERTIFICATES));
+    } else if (hasCertificateModes && certificates.length) {
+      dispatch(setMode(MODE_STATES.VIEW));
+    }
+  }, [hasCertificateModes, certificates]);
```
Comment (monteri) [inline src/certificates/hooks/useCertificates.jsx:36]:
```suggestion
useEffect(() => {
  if (!hasCertificateModes) {
    dispatch(setMode(MODE_STATES.NO_MODES));
  } else if (certificates.length === 0) {
    dispatch(setMode(MODE_STATES.NO_CERTIFICATES));
  } else {
    dispatch(setMode(MODE_STATES.VIEW));
  }
}, [hasCertificateModes, certificates]);
```

File: src/certificates/scss/Certificates.scss [hunk 9]
```diff
@@ -0,0 +1,75 @@
+.certificates {
+    h2, h3 {
+        color: $black;
+    }
+
+    .details-info {
```
Comment (monteri) [inline src/certificates/scss/Certificates.scss:6]:
```suggestion
    .certificates__details-info {
```

Comment (monteri) [reply src/certificates/scss/Certificates.scss:6]: and all classnames below

File: src/generic/modal-dropzone/useModalDropzone.jsx [hunk 10]
```diff
@@ -0,0 +1,83 @@
+import { useState } from 'react';
+import { useParams } from 'react-router-dom';
+import { useIntl } from '@edx/frontend-platform/i18n';
+
+import { uploadAssets } from './data/api';
+import messages from './messages';
+
+const useModalDropzone = ({ onChange, onCancel, onClose }) => {
```
Comment (monteri) [inline src/generic/modal-dropzone/useModalDropzone.jsx:8]: Allow to pass `fileTypes` `Array`

File: src/generic/modal-dropzone/useModalDropzone.jsx [hunk 11]
```diff
@@ -0,0 +1,83 @@
...
+        setPreviewUrl(reader.result);
+        setDisabledUploadBtn(false);
+      };
+      reader.readAsDataURL(file);
+      setSelectedFile(fileData);
+    }
+  };
+
+  const imageDimensionValidator = async (file) => {
+    if (file.type !== 'image/png') {
```
Comment (monteri) [inline src/generic/modal-dropzone/useModalDropzone.jsx:31]:
```suggestion
    if (!fileTypes.includes(file.type)) {
```

File: src/generic/modal-dropzone/ModalDropzone.test.jsx [hunk 12]
```diff
@@ -0,0 +1,83 @@
...
+    const uploadButton = getByText(messages.uploadModal.defaultMessage);
+    expect(uploadButton).toBeDisabled();
+  });
+
+  it('enables the upload button after a file is selected', async () => {
+    const { getByText, getByRole } = render(<RootWrapper {...props} />);
+    const dropzoneInput = getByRole('presentation');
+
+    await act(async () => {
+      fireEvent.change(dropzoneInput, { target: { files: ['test-file.png'] } });
```
Comment (monteri) [inline src/generic/modal-dropzone/ModalDropzone.test.jsx:76]: `fireEvent` change to `userEvent`

File: src/certificates/with-modes-without-certificates/WithModesWithoutCertificates.jsx [hunk 13]
```diff
@@ -1,34 +1,35 @@
...
 import { Button, Stack, Card } from '@edx/paragon';
 import { Add as AddIcon } from '@edx/paragon/icons';
 
-import MainLayout from '../layout/MainLayout';
+import { setMode } from '../data/slice';
 import messages from '../messages';
+import { MODE_STATES } from '../data/constants';
 
-const WithModesWithoutCertificates = ({ courseId }) => {
+const WithModesWithoutCertificates = () => {
```
Comment (monteri) [inline src/certificates/with-modes-without-certificates/WithModesWithoutCertificates.jsx:10]: `EmptyCertificatesWithModes`

File: src/certificates/Certificates.jsx [hunk 14]
```diff
@@ -1,23 +1,44 @@
...
+import CertificatesCards from './certificates-cards/CertificatesCards';
+import CertificateCreate from './certificate-create/CertificateCreate';
+import { MODE_STATES } from './data/constants';
+import MainLayout from './layout/MainLayout';
+
+const modeCertificatesMap = {
+  [MODE_STATES.NO_MODES]: WithoutModes,
+  [MODE_STATES.NO_CERTIFICATES]: WithModesWithoutCertificates,
+  [MODE_STATES.CREATE]: CertificateCreate,
+  [MODE_STATES.VIEW]: CertificatesCards,
```
Comment (monteri) [inline src/certificates/Certificates.jsx:17]:
```suggestion
  [MODE_STATES.NO_MODES]: CertificateWithoutModes,
  [MODE_STATES.NO_CERTIFICATES]: EmptyCertificatesWithModes,
  [MODE_STATES.CREATE]: CertificateCreate,
  [MODE_STATES.VIEW]: CertificatesCards,
```

**Review by monteri — CHANGES_REQUESTED** [review]
Please, fix lint issues

**Pull request closed without merging**